Using newer client versions in Ducktape tests

mmaslankaprv opened the pull request:
Updated Kafka tools and librdkafka to the new versions in our test environments. 

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

mmaslankaprv committed:
tests: updated kafka tools

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests: updated librdkafka

Signed-off-by: Michał Maślanka <michal@redpanda.com>

vbotbuildovich commented:
<!--NEW FAILURES-->
<!--070d78df-d7a9-4ce7-93ab-a88fbf692a5a-->
<!--NEW FAILURES-->
<!--eddae3e1-a2a5-4570-8006-c980c9ec52e4-->
<!--NEW FAILURES-->
<!--025a0552-d0b5-4c5d-b48b-5a8958281e44-->
<!--NEW FAILURES-->
<!--458104d9-a773-41aa-8e76-ff3ebd21805a-->
<!--NEW FAILURES-->
<!--3b402a3f-164b-4dae-81fa-d1daa04a20b3-->
<!--NEW FAILURES-->
<!--1a3be8cb-af1a-4bd4-a4be-4b4ec9f5028f-->
<!--NEW FAILURES-->
<!--70917195-4d19-4dd7-a725-bce38cb29b80-->
<!--NEW FAILURES-->
<!--4c46fc92-4c2d-4b61-b6bd-bcea5ed124c2-->
<!--NEW FAILURES-->
<!--2cd97a31-7142-448b-85c1-7b2cd24dedd0-->
<!--NEW FAILURES-->
<!--cc552ccc-52c5-40a5-93f9-5b053ca6e9f0-->
<!--NEW FAILURES-->
<!--c50358d6-547d-4c0e-b6e5-9a4451d3af41-->
<!--NEW FAILURES-->
<!--d89580fd-2c96-46bc-829d-c666f3620df9-->
<!--NEW FAILURES-->
<!--d654890c-4f77-42cf-b740-3665d1d1b551-->
<!--NEW FAILURES-->
<!--ab21b9cc-72c0-4112-881f-65f70f4e046b-->
<!--NEW FAILURES-->
<!--400486e7-17d3-47dc-b71e-75e17350a598-->
<!--NEW FAILURES-->
<!--6ebb1865-507c-4800-b2bf-55153110d24c-->
<!--NEW FAILURES-->
<!--74c32e76-e934-4cb2-8f06-708190d22482-->
<!--NEW FAILURES-->
<!--cd2d9234-6756-4490-b066-08ad1b50685a-->
<!--NEW FAILURES-->
<!--000108b1-242d-4847-8463-03f2bb49cf02-->
<!--NEW FAILURES-->
<!--9fa4b088-130e-4685-bb4b-df2967153b87-->
<!--body-->
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59593#0193b608-bae1-4797-9407-2926e537d9b9:
```
"rptest.tests.delete_records_test.DeleteRecordsKafkaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.path"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.fetch_after_deletion_test.FetchAfterDeleteTest.test_fetch_after_committed_offset_was_removed.transactions_enabled=False"
"rptest.tests.full_disk_test.LocalDiskReportTimeTest.test_target_min_capacity_wanted_time_based"
"rptest.tests.schema_registry_test.SchemaRegistryConfluentClient.test_versions"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_test.with_restart=True"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59593#0193b608-bae7-4220-9aa1-6c822b61a437:
```
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.ABS.2.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.kafka_cli_client_compat_test.KafkaCliClientCompatTest.test_create_topic"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59593#0193b608-bae4-4fb9-ae52-ebb2a488d724:
```
"rptest.tests.kafka_cli_client_compat_test.KafkaCliClientCompatTest.test_create_bad_acl"
"rptest.tests.delete_records_test.DeleteRecordsRedpandaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_connection_drop.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_timeboxed_uploads.acks=-1.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59593#0193b608-bae5-4166-b6f2-b0b044b04f5d:
```
"rptest.tests.kafka_cli_client_compat_test.KafkaCliClientCompatTest.test_create_role_acl"
"rptest.tests.archival_test.ArchivalTest.test_connection_drop.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_timeboxed_uploads.acks=-1.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_orphan_files_test"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59593#0193b5fa-0284-4a84-9539-3ae5db3c0685:
```
"rptest.tests.kafka_cli_client_compat_test.KafkaCliClientCompatTest.test_create_bad_acl"
"rptest.tests.delete_records_test.DeleteRecordsRedpandaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.path"
"rptest.tests.schema_registry_test.SchemaRegistryConfluentClient.test_versions"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_orphan_files_test"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59593#0193b5fa-0287-4e1e-9a5a-49bbe7550e3b:
```
"rptest.tests.delete_records_test.DeleteRecordsKafkaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.ABS.2.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.fetch_after_deletion_test.FetchAfterDeleteTest.test_fetch_after_committed_offset_was_removed.transactions_enabled=False"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a5-6809-43e2-85af-73f7809a4c49:
```
"rptest.tests.delete_records_test.DeleteRecordsRedpandaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_connection_drop.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_timeboxed_uploads.acks=-1.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a5-6807-45f1-9c31-803c17138aba:
```
"rptest.tests.delete_records_test.DeleteRecordsKafkaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.path"
"rptest.tests.fetch_after_deletion_test.FetchAfterDeleteTest.test_fetch_after_committed_offset_was_removed.transactions_enabled=False"
"rptest.tests.full_disk_test.LocalDiskReportTimeTest.test_target_min_capacity_wanted_time_based"
"rptest.tests.schema_registry_test.SchemaRegistryConfluentClient.test_versions"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_test.with_restart=True"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a5-680c-46e6-b1be-23b9adc47df0:
```
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.ABS.2.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a9-b307-4f5b-85b0-afd693f6aed7:
```
"rptest.tests.delete_records_test.DeleteRecordsRedpandaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.path"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.schema_registry_test.SchemaRegistryConfluentClient.test_versions"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_orphan_files_test"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a9-b30b-4072-8b9e-4c0e81173034:
```
"rptest.tests.delete_records_test.DeleteRecordsKafkaTest.test_delete_records_non_empty_topic.truncate_point=below_zero_not_negative_one"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_one_node_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.ABS.2.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_connection_flicker.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.fetch_after_deletion_test.FetchAfterDeleteTest.test_fetch_after_committed_offset_was_removed.transactions_enabled=False"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a5-680a-42c0-9c21-6b62695414c2:
```
"rptest.tests.archival_test.ArchivalTest.test_connection_drop.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_timeboxed_uploads.acks=-1.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_orphan_files_test"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a9-b30a-46b1-a496-e4ee860efa56:
```
"rptest.tests.archival_test.ArchivalTest.test_connection_drop.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_reconnect.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_timeboxed_uploads.acks=-1.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.topic_delete_test.TopicDeleteTest.topic_delete_test.with_restart=True"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a9-b309-4ba1-9785-34574dabe533:
```
"rptest.tests.archival_test.ArchivalTest.test_connection_drop.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_reconnect.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.archival_test.ArchivalTest.test_write.cloud_storage_type_and_url_style=.CloudStorageType.S3.1.virtual_host"
"rptest.tests.archival_test.ArchivalTest.test_timeboxed_uploads.acks=-1.cloud_storage_type=CloudStorageType.ABS"
"rptest.tests.full_disk_test.LocalDiskReportTimeTest.test_target_min_capacity_wanted_time_based"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59644#0193ba9d-ae99-4cc1-bc94-508e6ff014aa:
```
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59644#0193ba9d-ae9a-4deb-a24e-b313cea2a1be:
```
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59644#0193ba9d-ae97-4265-b9d8-88b818819c8e:
```
"rptest.tests.partition_reassignments_test.PartitionReassignmentsTest.test_reassignments_kafka_cli"
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.archival_test.ArchivalTest.test_single_partition_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59644#0193bab7-e3cb-4d65-a921-9bd92c00dafd:
```
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59644#0193bab7-e3c7-452a-a234-f71b1fb8f58e:
```
"rptest.tests.archival_test.ArchivalTest.test_all_partitions_leadership_transfer.cloud_storage_type=CloudStorageType.S3"
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59644#0193bab7-e3c9-40f4-9fd7-d6936bf4b914:
```
"rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3"
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59593-->
<!--b83ff5c7-e9ad-40c0-9254-993c70948f22-->
<!--8c7fb27a-db6a-4fc2-9b9e-3da1f4d95077-->
<!--b1195f9e-df6a-44f6-bff1-58fd134a5dee-->
<!--69376c21-ad4f-40c4-b32a-fc122f5ad975-->
<!--9c08b082-85b7-4f8f-9d70-847f2b82b41b-->
<!--818ca616-2a19-46cc-aec5-e36bb89759b3-->
<!--title-->
## Retry command for [Build#59593](https://buildkite.com/redpanda/redpanda/builds/59593)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/delete_records_test.py::DeleteRecordsKafkaTest.test_delete_records_non_empty_topic@{"truncate_point":"below_zero_not_negative_one"}
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_one_node_reconnect@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_single_partition_leadership_transfer@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_write@{"cloud_storage_type_and_url_style":[1,"path"]}
tests/rptest/tests/archival_test.py::ArchivalTest.test_connection_flicker@{"cloud_storage_type":1}
tests/rptest/tests/fetch_after_deletion_test.py::FetchAfterDeleteTest.test_fetch_after_committed_offset_was_removed@{"transactions_enabled":false}
tests/rptest/tests/full_disk_test.py::LocalDiskReportTimeTest.test_target_min_capacity_wanted_time_based
tests/rptest/tests/schema_registry_test.py::SchemaRegistryConfluentClient.test_versions
tests/rptest/tests/topic_delete_test.py::TopicDeleteTest.topic_delete_test@{"with_restart":true}
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_one_node_reconnect@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_write@{"cloud_storage_type_and_url_style":[2,"virtual_host"]}
tests/rptest/tests/archival_test.py::ArchivalTest.test_single_partition_leadership_transfer@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_connection_flicker@{"cloud_storage_type":2}
tests/rptest/tests/kafka_cli_client_compat_test.py::KafkaCliClientCompatTest.test_create_topic
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":2}
tests/rptest/tests/kafka_cli_client_compat_test.py::KafkaCliClientCompatTest.test_create_bad_acl
tests/rptest/tests/delete_records_test.py::DeleteRecordsRedpandaTest.test_delete_records_non_empty_topic@{"truncate_point":"below_zero_not_negative_one"}
tests/rptest/tests/archival_test.py::ArchivalTest.test_connection_drop@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_reconnect@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_write@{"cloud_storage_type_and_url_style":[1,"virtual_host"]}
tests/rptest/tests/archival_test.py::ArchivalTest.test_timeboxed_uploads@{"acks":-1,"cloud_storage_type":2}
tests/rptest/tests/kafka_cli_client_compat_test.py::KafkaCliClientCompatTest.test_create_role_acl
tests/rptest/tests/archival_test.py::ArchivalTest.test_connection_drop@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_reconnect@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_timeboxed_uploads@{"acks":-1,"cloud_storage_type":1}
tests/rptest/tests/topic_delete_test.py::TopicDeleteTest.topic_delete_orphan_files_test
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59601-->
<!--c4eba6f9-aa74-434a-b101-06289e4d9a9a-->
<!--title-->
## Retry command for [Build#59601](https://buildkite.com/redpanda/redpanda/builds/59601)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/delete_records_test.py::DeleteRecordsRedpandaTest.test_delete_records_non_empty_topic@{"truncate_point":"below_zero_not_negative_one"}
tests/rptest/tests/archival_test.py::ArchivalTest.test_connection_drop@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_reconnect@{"cloud_storage_type":2}
tests/rptest/tests/archival_test.py::ArchivalTest.test_write@{"cloud_storage_type_and_url_style":[1,"virtual_host"]}
tests/rptest/tests/archival_test.py::ArchivalTest.test_timeboxed_uploads@{"acks":-1,"cloud_storage_type":2}
```
<!--/body-->

vbotbuildovich commented:
<!--RETRIED FAILURES-->
<!--91790794-a7b6-4e14-acca-fbe656133d15-->
<!--RETRIED FAILURES-->
<!--a2f8e63f-45b5-4b9b-80e9-c23c9f4ec54c-->
<!--body-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59601#0193b6a9-b309-4ba1-9785-34574dabe533
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59644#0193bab7-e3c9-40f4-9fd7-d6936bf4b914
<!--/body-->

mmaslankaprv committed:
tests: exclude kafka internal topics from archival_test

Now, with the new Kafka tools the producer is by default using
idempotent produce requests. This creates an id allocator topic which
archival handles differently than usual kafka topics.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests/kafka_cli_tools: skip any additional log lines in output

The new version of Kafka CLI outputs additional log lines on error. We
need to skip them to correctly parse the result.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests: do not disable transactions in fetch after delete test

transactions are are required for idempotent producers to work
correctly.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests: recreate schema registry client when testing version

The schema registry client caches the schemas, recreating client drops
the caches and force it to request information from the registry.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
tests: skip internal topics when querying for kvstore items

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv committed:
test/kafka_cli_tools: added ability to disable idempotence

Signed-off-by: Michał Maślanka <michal@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59644-->
<!--95ab5763-cf94-487a-bf4d-b25ac05e5241-->
<!--63bde583-75bf-484b-a421-eea0cdd72341-->
<!--15afcf4a-4ed8-42a9-a4e1-89f518852683-->
<!--d409497e-02e2-48d1-b5e5-c9441993b498-->
<!--facaf131-eb2d-4563-85f3-c6dc9738ffbd-->
<!--e839d2b3-de03-42a4-8f68-fe4e464ab35c-->
<!--title-->
## Retry command for [Build#59644](https://buildkite.com/redpanda/redpanda/builds/59644)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":2}
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":1}
tests/rptest/tests/partition_reassignments_test.py::PartitionReassignmentsTest.test_reassignments_kafka_cli
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_single_partition_leadership_transfer@{"cloud_storage_type":1}
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer@{"cloud_storage_type":2}
```
<!--/body-->

michael-redpanda commented:
Is it possible that if we are only using the newest clients, we may start releasing code that causes issues with older clients?  I guess to maybe put it another way, does this bias our implementations and testing to ignore older clients?

mmaslankaprv commented:
> Is it possible that if we are only using the newest clients, we may start releasing code that causes issues with older clients? I guess to maybe put it another way, does this bias our implementations and testing to ignore older clients?

This question ca be asked both sides, i.e. do we ignore new clients while using the older ones ? We can and probably we should add basic tests that will sweep through the client versions and test some basic features.

michael-redpanda commented:
> > Is it possible that if we are only using the newest clients, we may start releasing code that causes issues with older clients? I guess to maybe put it another way, does this bias our implementations and testing to ignore older clients?
> 
> This question ca be asked both sides, i.e. do we ignore new clients while using the older ones ? We can and probably we should add basic tests that will sweep through the client versions and test some basic features.

Definitely not opposed to updating our test clients - absolutely should do that.  I wonder if, when a client is selected, we randomly select a version to use rather than always using the most recent one?

mmaslankaprv committed:
tests: increase the timeout in archival_test

Signed-off-by: Michał Maślanka <michal@redpanda.com>

mmaslankaprv commented:
/ci-repeat 1

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59864-->
<!--36ee1656-e576-4edd-8c96-7f627246348f-->
<!--dff92c40-3bca-440a-a22c-d11658e9d9c3-->
<!--86c62bc2-3da0-4ba3-97a8-31a448e16fcc-->
<!--36cdfad2-ee35-40ba-8f79-e71ede99dd36-->
<!--e8255fdf-7a45-4116-b711-9df2bcfce76a-->
<!--5368e936-2edb-4978-bef3-a7fec47a7d02-->
<!--title-->
## Retry command for [Build#59864](https://buildkite.com/redpanda/redpanda/builds/59864)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":2}
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":1}
tests/rptest/tests/partition_reassignments_test.py::PartitionReassignmentsTest.test_reassignments_kafka_cli
tests/rptest/tests/cloud_retention_test.py::CloudRetentionTest.test_cloud_retention@{"cloud_storage_type":2,"max_consume_rate_mb":20}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--bcae58c4-dc27-4339-b4c1-4bdf03bcf307-->
<!--TEST RESULTS-->
<!--29c4dce4-fa51-4e7f-87a1-360a23652a1d-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/59864">build#59864</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_cloud_retention.max_consume_rate_mb=20.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/59864#0193d5ae-1f3f-4de9-bf39-90ea92b7b0f9 | FAIL | 0/6 |
| rptest.tests.partition_reassignments_test.PartitionReassignmentsTest.test_reassignments_kafka_cli | ducktape | https://buildkite.com/redpanda/redpanda/builds/59864#0193d5a9-84a6-4479-9227-340d1ec6672e | FAIL | 0/1 |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/59864#0193d5a9-84a8-4cb6-a673-7f460326c0d4 | FAIL | 0/1 |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/59864#0193d5ae-1f3c-4e33-8afb-c11786ea3590 | FAIL | 0/1 |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/59864#0193d5a9-84a5-4eba-8a9f-93dac163a719 | FAIL | 0/1 |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/59864#0193d5ae-1f3d-4020-a56c-e9afc0b83f9f | FAIL | 0/1 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/60216">build#60216</a></summary>


| test_id                                                                                                                              | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| partition_balancer_planner_test_rpunit.partition_balancer_planner_test_rpunit                                                        | unit        | https://buildkite.com/redpanda/redpanda/builds/60216#019426a3-0895-4498-aee3-9b03f9a052ef | FLAKY         | 1/2      |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade                                | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60216#019426ff-610d-473b-ac07-6e7dab053b90 | FAIL          | 0/1      |
| rptest.tests.datalake.partition_movement_test.PartitionMovementTest.test_cross_core_movements.cloud_storage_type=CloudStorageType.S3 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60216#019426ff-610f-47a7-96b9-1f349a85ee9e | FLAKY         | 3/6      |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60216#019426fc-3bba-40ec-87b2-344aa5e0a086 | FAIL          | 0/1      |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.ABS                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60216#019426ff-610d-473b-ac07-6e7dab053b90 | FAIL          | 0/1      |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60216#019426fc-3bbb-4dca-b484-a1577ee548cc | FAIL          | 0/1      |
| rptest.tests.topic_recovery_test.TopicRecoveryTest.test_size_based_retention.cloud_storage_type=CloudStorageType.S3                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/60216#019426ff-610e-477a-9f55-e436843a28c8 | FAIL          | 0/1      |
</details>
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/60216-->
<!--dad32654-a84b-49e2-8f18-1854ec815c39-->
<!--baf5a0d2-cc23-4c30-9767-012c3d1c5232-->
<!--82725e05-bf1c-40f7-9d59-1ccafb2e5e42-->
<!--e65d40d4-7dcf-415f-86f9-9a48850b4ebd-->
<!--title-->
## Retry command for [Build#60216](https://buildkite.com/redpanda/redpanda/builds/60216)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":2}
tests/rptest/tests/topic_recovery_test.py::TopicRecoveryTest.test_size_based_retention@{"cloud_storage_type":1}
tests/rptest/tests/compaction_recovery_test.py::CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade
```
<!--/body-->

BenPope commented on tests/rptest/clients/kafka_cli_tools.py:
```diff
@@ -65,7 +65,8 @@ class KafkaCliTools:
     """
 
     # See tests/docker/Dockerfile to add new versions
```
This comment looks out of date.

BenPope commented on tests/rptest/tests/archival_test.py:
```diff
@@ -224,7 +224,7 @@ def __init__(self, test_context):
         self._s3_port = self.si_settings.cloud_storage_api_endpoint_port
 
-        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.kafka_tools = KafkaCliTools(self.redpanda, version="3.9.0")
```
This is currently default version, can you describe the intent of this change in the commit message, please

mmaslankaprv committed:
tests: exclude internal topics from recovery tests

Signed-off-by: Michał Maślanka <michal@redpanda.com>

BenPope reviewed (approved):
LGTM
